pkg/chains/signing_test.go

```diff
@@ -201,7 +204,74 @@ func TestTaskRunSigner_SignTaskRun(t *testing.T) {
 	}
 }
 
-func setupMocks(backends []*mockBackend) func() {
+func TestTaskRunSigner_Transparency(t *testing.T) {
+	rekor := &mockRekor{}
+	backends := []*mockBackend{{backendType: "mock"}}
+	cleanup := setupMocks(backends, rekor)
+	defer cleanup()
+
+	ctx, _ := rtesting.SetupFakeContext(t)
+	logger := logtesting.TestLogger(t)
+	ps := fakepipelineclient.Get(ctx)
+
+	cfgStore := &mockConfig{cfg: config.Config{
+		Artifacts: config.ArtifactConfigs{
+			TaskRuns: config.Artifact{
+				Format:         "tekton",
+				StorageBackend: "mock",
+				Signer:         "pgp",
+			},
+		},
+		Transparency: config.TransparencyConfig{
+			Enabled: false,
+		},
+	}}
+
+	ts := &TaskRunSigner{
+		Logger:            logger,
+		Pipelineclientset: ps,
+		SecretPath:        "./signing/pgp/testdata/",
+		ConfigStore:       cfgStore,
+	}
+
+	tr := &v1beta1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+		},
+	}
+	if _, err := ps.TektonV1beta1().TaskRuns(tr.Namespace).Create(ctx, tr, metav1.CreateOptions{}); err != nil {
+		t.Errorf("error creating fake taskrun: %v", err)
+	}
+	if err := ts.SignTaskRun(ctx, tr); err != nil {
+		t.Errorf("TaskRunSigner.SignTaskRun() error = %v", err)
+	}
+
+	if len(rekor.entries) != 0 {
+		t.Error("expected no transparency log entries!")
+	}
+
+	// Now enable and try again!
+	cfgStore.cfg.Transparency.Enabled = true
+	ts.ConfigStore = cfgStore
+
+	tr2 := &v1beta1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foobar",
+		},
+	}
+	if _, err := ps.TektonV1beta1().TaskRuns(tr.Namespace).Create(ctx, tr2, metav1.CreateOptions{}); err != nil {
+		t.Errorf("error creating fake taskrun: %v", err)
+	}
+	if err := ts.SignTaskRun(ctx, tr2); err != nil {
+		t.Errorf("TaskRunSigner.SignTaskRun() error = %v", err)
+	}
+
+	if len(rekor.entries) != 1 {
```

Comment:
i think i'm missing something here, but shouldn't this still be 0 since the `pgp` signer was configured above??

Reply:
Heh, there was some trickery. It should be all correct now.